src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -412,11 +412,11 @@ subroutine Infiltration(bounds, num_hydrologyc, filter_hydrologyc, num_urbanc, f
 
              !--  if all water evaporates, there will be no bottom drainage
              if (h2osfc(c) < 0.0) then
-                qflx_infl(c) = qflx_infl(c) + h2osfc(c)/dtime
+                qflx_infl(c) = qflx_infl(c) + frac_h2osfc(c)*h2osfc(c)/dtime
                 h2osfc(c) = 0.0
                 qflx_h2osfc_drain(c)= 0._r8
              else
-                qflx_h2osfc_drain(c)=min(frac_h2osfc(c)*qinmax,h2osfc(c)/dtime)
+                qflx_h2osfc_drain(c)=min(qinmax, h2osfc(c)/dtime)
```

Comment:
**Comment from @martynpclark Aug 17, 2017**

Note here we are treating the ponded water as a 1-d column, so no sub-grid fractions are required.